plugins/jobs/src/js/components/JobsBreadcrumbs.js

```diff
@@ -4,62 +4,45 @@ import PropTypes from "prop-types";
 import React from "react";
 import { Tooltip } from "reactjs-components";
 
-import Job from "#SRC/js/structs/Job";
-import JobTree from "#SRC/js/structs/JobTree";
 import TaskStates from "#PLUGINS/services/src/js/constants/TaskStates";
 import Breadcrumb from "#SRC/js/components/Breadcrumb";
 import BreadcrumbSupplementalContent from "#SRC/js/components/BreadcrumbSupplementalContent";
 import BreadcrumbTextContent from "#SRC/js/components/BreadcrumbTextContent";
 import Icon from "#SRC/js/components/Icon";
 import PageHeaderBreadcrumbs from "#SRC/js/components/PageHeaderBreadcrumbs";
 
-function getLongestRunningTask(item) {
-  if (!(item instanceof Job)) {
+function ItemStatus({ item: { jobRuns } }) {
+  if (!jobRuns) {
     return null;
   }
-
-  const longestRunningActiveRun = item
-    .getActiveRuns()
-    .getLongestRunningActiveRun();
-
-  if (!longestRunningActiveRun) {
-    return null;
-  }
-
-  return longestRunningActiveRun.getTasks().getLongestRunningTask();
-}
-
-function getItemStatus(item) {
-  const longestRunningTask = getLongestRunningTask(item);
-
-  if (!longestRunningTask) {
+  const { longestRunningActiveRun } = jobRuns;
+  if (
+    !longestRunningActiveRun ||
+    !longestRunningActiveRun.tasks.longestRunningTask
+  ) {
     return null;
   }
+  const { status } = longestRunningActiveRun.tasks.longestRunningTask;
 
   return (
     <BreadcrumbSupplementalContent>
       <div className="service-page-header-status muted">
-        ({TaskStates[longestRunningTask.getStatus()].displayName})
+        ({TaskStates[status].displayName})
       </div>
     </BreadcrumbSupplementalContent>
   );
 }
 
-function getItemSchedule(item) {
-  if (!(item instanceof Job)) {
-    return null;
-  }
-
-  const schedule = item.getSchedules()[0];
-
-  if (!schedule || !schedule.enabled) {
+function ItemSchedule({ item: { schedules } }) {
```

Comment:
🎨  same here, it should no know that much about schedules and nodes, but receive the schedule node it needs to rendered, and if null do nothing.

Reply:
I understand that being out of the scope of this refactor as well